### PR TITLE
Build docker images with new tags

### PR DESCRIPTION
**Pull request by aerosol**
### Changes
 
This adds a new GitHub Action that runs on new tag creation. The workflow should result with a tagged docker image.

### Tests
- [ ] Automated tests have been added
- [x] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [x] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [x] This change does not need a documentation update

### Dark mode
- [ ] The UI has been tested both in dark and light mode
- [x] This PR does not change the UI

### CODE REVIEW

File: .github/workflows/build-tagged-images.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Build Tagged Images
+
+on:
+  push:
+      tags:
+          - '**'
```
Comment (vinibrsl) [inline .github/workflows/build-tagged-images.yml:6]: I'm not sure, but I think adding this to `.github/workflows/build-private-images.yml` and putting `type=ref,event=tag` up top would do it all in a single file.

Happy to go with this file, but just checking if you considered that.

**Review by vinibrsl — APPROVED** [review]

File: .github/workflows/build-tagged-images.yml (hunk 1 above)
Comment (ukutaht) [reply .github/workflows/build-tagged-images.yml:6]: Yes, I think extending `.github/workflows/build-private-images.yml` is a good idea unless there's some reason to keep this separate.

Comment (aerosol) [reply .github/workflows/build-tagged-images.yml:6]: > unless there's some reason to keep this separate.

Not at the moment, no. On it.

Comment (aerosol) [reply .github/workflows/build-tagged-images.yml]: OK I went with `r*` tagging scheme for now as discussed with @ukutaht elsewhere.

https://github.com/plausible/analytics/pull/2212/commits/0d2ae743d3016b2cf2a40be9a094858fdb207aa9 - if you think that'll work, I'll be happy to give it a go

**Comment by ukutaht** [comment]
I was going to do it myself but maybe better to flag: unless there's a reason to keep the branch around, let's delete it after the PR is merged to keep the nr of branches in the whole git tree reasonable.

**Comment by aerosol** [comment]
I just did, was mine after all! Apologies

**Pull request merged**